minimum_jerk_ros/src/minimum_jerk_ros.cpp

```diff
@@ -0,0 +1,736 @@
+        this->action_translation_server_ = std::make_shared<ActionServerTrans>(get_node_base_interface(),
+                                                                               get_node_clock_interface(),
+                                                                               get_node_logging_interface(),
+                                                                               get_node_waitables_interface(),
+                                                                               "translate", std::bind(&MinimumJerkRos::translation_callback, this));
+
+        collision_client_node_ = rclcpp::Node::make_shared("collision_client_node");
+        collision_srv_client_ = collision_client_node_->create_client<cmr_msgs::srv::IsCollisionFree>("is_collision_free");
+
+        obstacles_marker_pub_ = this->create_publisher<visualization_msgs::msg::Marker>("minimum_jerk_obstacle_markers", 1);
+        trajectory_pub_ = this->create_publisher<nav_msgs::msg::Path>("minimum_jerk_trajectory", 1);
+        goal_pose_pub_ = this->create_publisher<geometry_msgs::msg::PoseStamped>("minimum_jerk_goal_pose", 1);
+
+        obstacle_lookahead_distance_ = std::clamp(obstacle_lookahead_distance_, robot_radius_ + 0.1, 1.0);
+        return rclcpp_lifecycle::node_interfaces::LifecycleNodeInterface::CallbackReturn::SUCCESS;
+    }
+
+    rclcpp_lifecycle::node_interfaces::LifecycleNodeInterface::CallbackReturn MinimumJerkRos::on_activate(const rclcpp_lifecycle::State &)
+    {
+        RCLCPP_INFO(this->get_logger(), "on_activate()...");
+        this->action_rotation_server_->activate();
+        this->action_translation_server_->activate();
+        this->publisher_->on_activate();
+        this->obstacles_marker_pub_->on_activate();
+        this->goal_pose_pub_->on_activate();
+        this->trajectory_pub_->on_activate();
+        RCLCPP_INFO(this->get_logger(), "Action server successfully activated...");
+        return rclcpp_lifecycle::node_interfaces::LifecycleNodeInterface::CallbackReturn::SUCCESS;
+    }
+
+    rclcpp_lifecycle::node_interfaces::LifecycleNodeInterface::CallbackReturn MinimumJerkRos::on_deactivate(const rclcpp_lifecycle::State &)
+    {
+        RCLCPP_INFO(this->get_logger(), "on_deactivate()...");
+        this->action_rotation_server_->deactivate();
+        this->action_translation_server_->deactivate();
+        this->publisher_->on_deactivate();
+        this->obstacles_marker_pub_->on_deactivate();
+        this->goal_pose_pub_->on_deactivate();
+        this->trajectory_pub_->on_deactivate();
+        return rclcpp_lifecycle::node_interfaces::LifecycleNodeInterface::CallbackReturn::SUCCESS;
+    }
+
+    rclcpp_lifecycle::node_interfaces::LifecycleNodeInterface::CallbackReturn MinimumJerkRos::on_cleanup(const rclcpp_lifecycle::State &)
+    {
+        RCLCPP_INFO(this->get_logger(), "on_cleanup()...");
```

Comment:
Should test:

`ros2 launch ...`
`ros2 lifecycle set /nodename deactivate`
`ros2 lifecycle set /nodename cleanup`
`ros2 lifecycle set /nodename configure`
`ros2 lifecycle set /nodename activate`

To make sure you can transition down and up the node without any weird behaviours